nx_parallel/algorithms/shortest_paths/weighted.py

```diff
@@ -1,8 +1,87 @@
+# TODO : update docstrings
+
+"""
+Shortest path parallel algorithms for weighted graphs.
+"""
+
 from joblib import Parallel, delayed
-from networkx.algorithms.shortest_paths.weighted import single_source_bellman_ford_path
 import nx_parallel as nxp
+from networkx.algorithms.shortest_paths.weighted import (
+    single_source_dijkstra,
+    single_source_dijkstra_path_length,
+    single_source_dijkstra_path,
+    single_source_bellman_ford_path,
+    single_source_bellman_ford_path_length,
+)
+
+__all__ = [
+    "all_pairs_dijkstra",
+    "all_pairs_dijkstra_path_length",
+    "all_pairs_dijkstra_path",
+    "all_pairs_bellman_ford_path_length",
+    "all_pairs_bellman_ford_path",
+]
+
+
+def all_pairs_dijkstra(G, cutoff=None, weight="weight"):
+    """Find shortest weighted paths and lengths between all nodes."""
+
+    def _calculate_all_pairs_dijkstra_subset(n):
+        return (n, single_source_dijkstra(G, n, cutoff=cutoff, weight=weight))
+
+    if hasattr(G, "graph_object"):
+        G = G.graph_object
 
-__all__ = ["all_pairs_bellman_ford_path"]
+    return Parallel(n_jobs=-1, return_as="generator")(
+        delayed(_calculate_all_pairs_dijkstra_subset)(n) for n in G
+    )
+
+
+def all_pairs_dijkstra_path_length(G, cutoff=None, weight="weight"):
+    """Compute shortest path lengths between all nodes in a weighted graph."""
+    if hasattr(G, "graph_object"):
+        G = G.graph_object
+
+    length = single_source_dijkstra_path_length
+
+    def _calculate_all_pairs_dijkstra_path_length_subset(n):
+        return (n, length(G, n, cutoff=cutoff, weight=weight))
+
+    return Parallel(n_jobs=-1, return_as="generator")(
+        delayed(_calculate_all_pairs_dijkstra_path_length_subset)(n) for n in G
+    )
+
+
+def all_pairs_dijkstra_path(G, cutoff=None, weight="weight"):
+    """Compute shortest paths between all nodes in a weighted graph."""
+    if hasattr(G, "graph_object"):
+        G = G.graph_object
+
+    path = single_source_dijkstra_path
+
+    def _calculate_all_pairs_dijkstra_path_subset(n):
+        return (n, path(G, n, cutoff=cutoff, weight=weight))
```

Comment:
Same comments in dijkstra as above regarding namespace for the function name and input args when they don't depend on `n`.